### ISSUE TITLE
`bun --compile` error. `error: Could not find module root given file.`

### ISSUE THREAD

**Issue by zachcp** [oven-sh/bun#10964]
### What version of Bun is running?

1.1.7+b0b7db5c0

### What platform is your computer?

Darwin 23.4.0 x86_64 i386

### What steps can reproduce the bug?

```sh
# I'm trying to compile the molrender script within molstar
#
# Clone Mol* repo, install, and build
git clone https://github.com/molstar/molstar.git
cd molstar/
npm add gl@6.0.2
npm install && npm run build

# Render a MolViewSpec file `examples/mvs/1cbs.mvsj` to `../outputs/1cbs.png`
node lib/commonjs/cli/mvs/mvs-render.js -i examples/mvs/1cbs.mvsj -o ../outputs/1cbs.png --size 800x600 --molj

bun build lib/commonjs/cli/mvs/mvs-render.js --compile --outfile mol-render
```

### What is the expected behavior?

Creation of a usable cli program.

### What do you see instead?

```sh
❯ ./mol-render
4667 |       }
4668 |       if (exists(join(dir, "package.json")) || exists(join(dir, "node_modules"))) {
4669 |         return dir;
4670 |       }
4671 |       if (prev === dir) {
4672 |         throw new Error('Could not find module root given file: "' + file + '". Do you have a `package.json` file? ');
                     ^
error: Could not find module root given file: "/$bunfs/root/mol-render". Do you have a `package.json` file?
      at getRoot (/$bunfs/root/mol-render:4672:15)
      at bindings (/$bunfs/root/mol-render:4566:26)
      at /$bunfs/root/mol-render:4682:21
      at /$bunfs/root/mol-render:3:75
      at /$bunfs/root/mol-render:5888:58
      at /$bunfs/root/mol-render:3:75
      at /$bunfs/root/mol-render:8972:72
      at /$bunfs/root/mol-render:3:75
      at /$bunfs/root/mol-render:8984:22
      at /$bunfs/root/mol-render:3:75
      at /$bunfs/root/mol-render:161897:38
      at /$bunfs/root/mol-render:3:75
      at /$bunfs/root/mol-render:161919:16

Bun v1.1.7 (macOS x64)
```

### Additional information

- the error message is about the package.json
- the script I am trying to run is in a subdirectory.
- is there an additional parameter to handle this?

**Comment by ElGoru** [comment]
I have the same error, where you able to fix it?

**Comment by zachcp** [comment]
No

**Comment by Jarred-Sumner** [comment]
I guess we could copy the package.json into the compiled executable

**Comment by buzzy** [comment]
Seems this one is still not fixed. I get the same issue. https://github.com/oven-sh/bun/issues/14301

**Comment by pixer415** [comment]
> I guess we could copy the package.json into the compiled executable

I'm _**very**_ new to Bun. How exactly would I do this?

**Comment by Winterreisender** [comment]
I'm trying to fix the same error in my project and I find that it's related to [node-bindings](https://www.npmjs.com/package/bindings)

It's a widely used package to glue C++ Node-API addons to JS. And by default it will use `packages.json`'s locations to be the base folder to find the `.node` files, maybe the `process.cwd` or `exisits` is different inside and outside the executable file. It is incompatible with bun's executable packager (and other packagers?)

Here's the code of `bindings` that make the error:

```js
168 |       if (exists(join(dir, "package.json")) || exists(join(dir, "node_modules"))) {
169 |         return dir;
170 |       }
171 |         throw new Error('Could not find module root given file: "' + file + '". Do you have a `package.json` file? ');
```

But actually that the files

I don't know how the files are copied or emulated in bun, but here's a solution that works for me: 
Set the `module_root` to `'xxx'`(the package name) in the location that `bindings` is called. So that the `getRoot` won't be called
```js
// From this
    exports.libwebview = require('bindings')('xxx')
// To this
    exports.libwebview = require('bindings')({
        bindings: 'xxx.node',
        module_root: 'xxx'
}
```

I also don't know if this only happens in bun or both bun and node. if is the latter, this should be a bug of node-bindings

**Comment by m10rten** [comment]
Having this same issue while compiling a nestjs(ts) project, any ways to solve this bug?

**Comment by nickhobbs94** [comment]
I'm also seeing this error trying to compile the hello world example from `blessedjs` https://github.com/chjj/blessed?tab=readme-ov-file#example

**Comment by april83c** [comment]
Also running into this error when trying to run a compiled version of my project. Would love to see it fixed. Not as easy to work around it for me, as it's one of my dependencies using the `bindings` package, not me directly. 

Code that's throwing, in case it's helpful: https://github.com/TooTallNate/node-bindings/blob/c8033dcfc04c34397384e23f7399a30e6c13830d/bindings.js#L214

**Comment by suchipi** [comment]
I was able to work around this issue by using [patch-package](https://www.npmjs.com/package/patch-package) to replace all `require('bindings')('whatever')` in my node_modules with `require('./build/Release/whatever.node')` (for varying values of `whatever`). The replacement path may vary depending on the package you're loading, but the general idea is to look for a `.node` file in the package and require that directly instead of via `bindings`.

It'd be nice if bun did this automatically at bundle time, using a macro or etc.

**Comment by ctjlewis** [comment]
Yup, thanks so much @suchipi. @april83c, for `speaker` you can just do a crude string replace even:

```js
const binding = require('bindings')('binding')
```

to:

```js
const binding = require('./build/Release/binding.node')
```

And it will build, run, and the speaker works (just checked on Apple Silicon). cc @Jarred-Sumner on this one, another node-gyp related thing. `speaker` package was the same one for #23070.
